### ISSUE TITLE
WSL --mount will not mount a microSD card

### ISSUE THREAD

**Issue by dustojnikhummer** [microsoft/WSL#8090]
### Version

22000.527

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.10.16.3

### Distro Version

Ubuntu 20.04

### Repro Steps

Surface Go 1 (Yes I'm aware it doesn't officially support Windows 11)

Insert an unpartitioned microSD card
Run GET-CimInstance -query "SELECT * from Win32_DiskDrive"
wsl --mount \\.\PHYSICALDRIVE1

Screenshot
https://i.imgur.com/TqfUBoZ.png

### Expected Behavior

SD card gets mounted into WSL2

### Actual Behavior

The system cannot find the drive specified

**Comment by 0xbadfca11** [comment]
wsl --mount feature does not support removable media, use USP/IP instead.

> If you are looking for guidance on how to connect a USB device (flash drive, SD card reader, etc), see [Connect USB devices](https://docs.microsoft.com/en-us/windows/wsl/connect-usb).
> https://docs.microsoft.com/en-us/windows/wsl/wsl2-mount-disk
> https://github.com/MicrosoftDocs/WSL/commit/ea7499d84f3493590300495f003f84ffc36ffff0

**Comment by dustojnikhummer** [comment]
I did try that before leaving this report. That does not work either. The SD card just doesn't show up in the list. And not sure if it matters, but it isn't a USB SD card reader, but a PCIe one

https://i.imgur.com/XuYNZJk.png

**Comment by andyearnshaw** [comment]
I have the same issue on my laptop. SD card reader is not in the USB device list, and I can't mount it.

**Comment by SeWieland** [comment]
This is due to the fact, that there are simply no USB storage drivers in the WSL Kernel.
You can use [usbipd for connectiong any USB devices to WSL](https://learn.microsoft.com/en-us/windows/wsl/connect-usb) and they will also show up under `lsusb`, however due to the missing drivers in the kernel, no devices will be listed under `lsblk`.

I'm now trying to compile my own kernel according to the [usbipd docs](https://github.com/dorssel/usbipd-win/wiki/WSL-support#building-your-own-usbip-enabled-wsl-2-kernel), but it's probably faster to just flash ubuntu over your windows machine at this point.

I have really no idea why those drivers are not simply compiled into the default kernel.

See also many other issues like https://github.com/microsoft/WSL/issues/8408

**Comment by TurboVin** [comment]
I am still running into this issue for the Broadcom SD Card reader built into my old Acer laptop. 

`wsl --mount \.\PHYSICALDRIVE2` reports no drive found. 
device appears as SCSI Disk Drive in Device Manager
`usbipd wsl list` does not list the SD Card.

**Comment by ErencanPelin** [comment]
> I am still running into this issue for the Broadcom SD Card reader built into my old Acer laptop.
> 
> `wsl --mount \.\PHYSICALDRIVE2` reports no drive found. device appears as SCSI Disk Drive in Device Manager `usbipd wsl list` does not list the SD Card.

i'm in the same boat now, did you ever find a fix?

**Comment by KhalilOuali** [comment]
Same issue. I need the feature to access an SD card for a Raspberry Pi and modify some files, but it's not working :(

**Comment by sprsr** [comment]
Still no support for USB Storage in 2024 it seems.

**Comment by ssg** [comment]
The only working solution is to recompile the WSL kernel. It's less painful than you think. The full instructions are here: https://www.tomshardware.com/how-to/access-linux-ext4-partitions-in-windows

In addition to the instructions above, in order to read Steam Deck SD Cards, you must also enable casefolding support which requires UTF-8 character handling Under "File Systems -> UTF-9 normalization and casefolding support".

I hope Microsoft adds USB storage support to newer kernels.

**Comment by ccoles146** [comment]
This may be the same issue, I've got more detail in my issue: https://github.com/microsoft/WSL/issues/12153

**Comment by Ircama** [comment]
Until this limitation (maybe Hyper-V related?) is resolved, check out some possible [workarounds here](https://superuser.com/a/1869821/2523402).

**Comment by flowirin** [comment]
same issue. windows 10, wsl, raspberry pi formatted sd car on USB-C adapter.

**Comment by pinballelectronica** [comment]
I don't there are solutions but they're hacky outside of buying the right reader- I don't think recompiling the kernel is necessary.

Some usb adapters treat the sdcard as a drvfs filesystem (mass storage vs removable media).  That would solve the problem I believe.  My lexar sd/microsd reader treats them as drives and not removable media.

I know people hate to hear this but WSL1 may provide better access to drives that are identified as removable media.  WSL1 uses api calls to the fs which makes it a lot more compatible w/o hyper-v because there's no additional virtualization layer.  You'll see a lot of posts of people downgrading to WSL1.

Odd but I really like wsl2.

**Comment by hrply** [comment]
> The only working solution is to recompile the WSL kernel (5.15.153.1 currently). It's less painful than you think. The full instructions are here: https://www.tomshardware.com/how-to/access-linux-ext4-partitions-in-windows
> 
> In addition to the instructions above, in order to read Steam Deck SD Cards, you must also enable casefolding support which requires UTF-8 character handling Under "File Systems -> UTF-8 normalization and casefolding support".
> 
> I hope Microsoft adds USB storage support to newer kernels.

thanks~~but it not works for hardisk like nvme T-T

**Comment by SteffenSH** [comment]
This issue is a duplicate of to be a duplicate of #6011.

**Comment by vincentlarkin** [comment]
> The only working solution is to recompile the WSL kernel (5.15.153.1 currently). It's less painful than you think. The full instructions are here: https://www.tomshardware.com/how-to/access-linux-ext4-partitions-in-windows
> 
> In addition to the instructions above, in order to read Steam Deck SD Cards, you must also enable casefolding support which requires UTF-8 character handling Under "File Systems -> UTF-8 normalization and casefolding support".
> 
> I hope Microsoft adds USB storage support to newer kernels.

wow, what an insane thing you have to do to access the steam deck sd card on Windows. MS really dropped the ball here.

**Comment by ssg** [comment]
What a bad, nonsense policy to close this automatically due to lack of inactivity. The issue is still there, but you want to have a new copy of it without the information we helped collect here because we had the manners and the courtesy not to post "Any updates?" every week?

What you're actually doing is to push us more towards an archaic workaround like compiling our own WSL kernel which is neither efficient nor safe, and may generate more problems and more frustration along the way.

Even if this issue gets filed again in the future with more urgency because Valve is now releasing new Steam devices with SD card support, and WSL is currently the only way to access those file systems on Windows, now you're dooming the quality of the fix or maybe the test time for it because the feedback has to start from scratch now.

If you want to *prioritize* issues based on recent activity, just add a "last update" filter? You throw away all the volunteer work of people who have tried to investigate this because one person can't be bothered to do that? Do you see how disrespectful this looks?

Open source work is pretty much thankless, I know, you think you need some hard measures to make the firehose of issues workable, but what you're actually doing to alleviate it is harming your relationship with the community and the quality of your product. 

Don't let the future of this fantastic project be dictated by a bot. Bring back human.

**Comment by SteffenSH** [comment]
@ssg, I agree with you that it's disrespectful to just close issues like this automatically. That being said though, as for this issue in particular there already exists a couple of duplicates, one of which is still open and have some recent activity from some commenters.
That issue is here: #6011
Hopefully it will survive into the future or be fixed soon.

**Comment by pinballelectronica** [comment]
Seriously though, did you just refer to a 5+ year old thread and use the
word "recent" in the same breath?  The commenters are wondering what's
taking so long.  Anyone that thinks Microsoft has an interest in
perpetuating *nix I mean..Those poor WSL devs are probably sucking wind.

**Comment by SteffenSH** [comment]
@pinballelectronica, I was seriously referring to the [comment](https://github.com/microsoft/WSL/issues/6011#issuecomment-4757691180) on #6011 that was less than a week old. As far as I know that activity will prevent #6011 from being closed automatically like this one did. Also note that #6011 is the earliest duplicate so its only natural to keep that one open while closing the others (including this one).

Btw, did you know that Microsoft open sourced WSL some time ago. So in theory it is possible for anyone with the necessary skills and time to make an implementation for this. But nobody has attempted yet afaik.

**Comment by pinballelectronica** [comment]
Fair. I did not catch the they open sourced it in 2025, no.  There’s a lot
of closed-source Microsoft stuff in line with WSL though, which could end
up being a dead end when trying to work a MicroSD issue. I think a LOT of
the complaints are a result of this.  My guess is if it were purely in the
open WSL/Linux-side code, someone may already have fixed it or there may
already be a PR. I haven’t checked for a working PR though.

For storage, wouldn’t the Windows-side filesystem bridge come into play?

e.g. WSL DrvFs <-> Windows filesystem bridge <-> Windows removable-drive
stack

AFAIK, at least the Windows-side DrvFs / filesystem bridge pieces are not
fully open.

**Comment by SteffenSH** [comment]
I don't know. You're maybe correct about that :-)
